@uportal/grid-carousel/src/grid-carousel.js

```diff
@@ -0,0 +1,295 @@
+import { LitElement, css, html, unsafeCSS } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+import { portletRegistryToArray } from '@uportal/portlet-registry-to-array';
+import { register } from 'swiper/element/bundle';
+import swiperStyles from 'swiper/swiper.css?inline';
+import get from 'lodash/get';
+import oidc from '@uportal/open-id-connect/src/open-id-connect.js';
+
+@customElement('grid-carousel')
+class GridCarousel extends LitElement {
+  @property({ type: String, attribute: 'data-api-url' })
+  dataApiUrl = '/uPortal/api/v4-3/dlm/portletRegistry.json?favorite=true';
+
+  @property({ type: String, attribute: 'oidc-url' })
+  oidcUrl = '/uPortal/api/v5-1/userinfo';
+
+  @property({ type: String, attribute: 'context-portlet-url' })
+  contextPortletUrl = '/uPortal/p/';
+
+  @property({ type: Boolean })
+  debug = false;
+
+  @property({ type: String, attribute: 'default-icon' })
+  defaultIcon =
+    '/ResourceServingWebapp/rs/tango/0.8.90/32x32/categories/applications-other.png';
+
+  @property({ type: Number, attribute: 'truncate-length' })
+  truncateLength = 50;
+
+  @property({ type: Number, attribute: 'slides-per-view' })
+  slidesPerView = 3;
+
+  @property({ type: Number, attribute: 'slides-per-group' })
+  slidesPerGroup = 3;
+
+  @property({ type: Number })
+  speed = 200;
+
+  @property({ type: Number, attribute: 'space-between' })
+  spaceBetween = 20;
+
+  @property({ type: Number })
+  breakpoint = 768;
+
+  @state({ type: Boolean })
+  mobile = false;
+
+  @state({ type: Array })
+  items = [];
+
+  @state({ type: String })
+  errorMessage = null;
+
+  constructor() {
+    super();
+  }
+
+  truncateTitle(title) {
+    if (title.length > this.truncateLength) {
+      return title.substring(0, this.truncateLength + 1) + '...';
+    }
+
+    return title;
+  }
+
+  mapRegistry(registry) {
+    return registry.map(({ fname, parameters, title }) => {
+      const imgUrl =
+        get(parameters, 'waffleIconUrl.value') ||
+        get(parameters, 'iconUrl.value') ||
+        this.defaultIcon;
+      const text = get(parameters, 'waffleLabel.value') || title;
+
+      const alternativeMaximizedLink = get(
+        parameters,
+        'alternativeMaximizedLink.value'
+      );
+
+      let targetLinkValue = '_self';
+      if (alternativeMaximizedLink) {
+        targetLinkValue = '_blank';
+      }
+
+      return {
+        id: fname,
+        link: alternativeMaximizedLink || this.contextPortletUrl + fname,
+        image: imgUrl ? imgUrl : this.defaultIcon,
```

Comment:
`imgUrl` would be undefined?
If so would:

```suggestion
        image: imgUrl ?? this.defaultIcon,
```

make sense?